packages/core/src/components/tooltip/tooltip.tsx

```diff
@@ -134,6 +177,14 @@ export class Tooltip<
                     },
                 }}
                 {...restProps}
+                renderTarget={
+                    renderTarget &&
+                    React.useCallback(props => renderTarget({ ...props, tooltipId }), [renderTarget, tooltipId])
```

Comment:
Similar to above, can we pull this `React.useCallback` into a variable at the top of the component such that it is not called conditionally?